Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/Model/Uploads/UploadsManager.swift

```diff
@@ -102,9 +102,46 @@ class UploadsManager: ObservableObject {
         performUpload(upload)
     }
     
+    @MainActor func uploadRawMapboxData(inTemporaryFile temporaryFile: TemporaryFile) {
+        guard s3Helper != nil else {
+            logger.info("Skipping upload of raw Mapbox data since S3 is not configured")
+            return
+        }
+        
+        let uploadId = UUID()
+        let dataFileURL: URL
+        do {
+            let storageDirectoryURL = try Storage.storageDirectoryURL(forUploadId: uploadId.uuidString)
+            try Storage.ensureStorageDirectoryExists(atURL: storageDirectoryURL)
+            dataFileURL = Storage.dataURL(forStorageDirectoryURL: storageDirectoryURL)
+            
+            try temporaryFile.stayAlive {
```

Comment:
Could you elaborate on why you went with the "stayAlive" block approach?

Couldn't we just remove that file after the upload to S3 is successful? (I imagine that's when we want to clean it up?)

Right now it's kinda difficult for me to understand when the `temporaryFile` is expected to actually deallocate and remove the file underneath.

Reply:
The problem is that it's the SDK that emits the URL of the file, but it's the example app that uploads to S3. So the SDK needs a way to know that the consumer of the file is finished with it, so that it can clear the file up. The responsibility for clearing up the file needs to live with the SDK, because there's no guarantee that the user of the SDK is even going to _use_ this file, so it can't be expected to clean it up.

Reply:
> Right now it's kinda difficult for me to understand when the `temporaryFile` is expected to actually deallocate and remove the file underneath.

Do you think it's important for a user of the SDK to understand this? The point is that as long as they hold on to the `TemporaryFile` object, they have access to the file. And then at some point after they let go of it, the SDK will take care of clearing it up.

Reply:
Good point 👍